### PR TITLE
Add parent and type attributes for statements and Fix #2729 and Fix #2694

### LINKED ISSUE #2694: Break doesn't call finally block.
See following example,
```
function main(string[] args){
int i = 0;
    while (i < 10) {
        i = i + 1;
        try {
            if (i == 5) {
                break;
            }
        }finally {
             system:println("Finally Block executed." + i);
        }
    }
}
```
getting following result. 
Finally Block executed.1
Finally Block executed.2
Finally Block executed.3
Finally Block executed.4

But it should have 
Finally Block executed.5 as well.

### LINKED ISSUE #2729: Restricting Break/Continue in a transaction.
Currently, there is a restriction, where you can't use a return in a transaction statement. Similarly, we have to enforce the same rule for the Break and continue as well, but​ ​if​ only​ those statements used directly inside a transaction. 

Eg:​ ​This shouldn't be allowed

function foo () {
    int i = 0;
    while (i < 5) {
​​        i = i + 1;​
        transaction {
            if (i == 2) {
               ​ break;​// Or continue​;  
            }
        }
    }
}

But t​his is Ok. 
​
function foo () {
    int i = 0;
    transaction {
        while (i < 5) {
            i = i + 1;
            if (i == 2) {
                break;
            }
        }
    }
}​

### PR DESCRIPTION
**Pull request by hasithaa**

### CODE REVIEW

File: modules/ballerina-core/src/main/java/org/ballerinalang/model/statements/AbstractStatement.java [hunk 1]
```diff
@@ -58,4 +60,24 @@ public void setAlwaysReturns(boolean alwaysReturns) {
...
+        return this.parent;
+    }
+
+    @Override
+    public void setType(StatementType statementType) {
+        this.type = statementType;
+    }
+
+    @Override
+    public StatementType getType() {
```
Comment (sameera-jayasoma) [inline modules/ballerina-core/src/main/java/org/ballerinalang/model/statements/AbstractStatement.java]: This getType() method will confuse developers with getType() usages in the other parts of the code.

Comment (sameera-jayasoma) [inline modules/ballerina-core/src/main/java/org/ballerinalang/model/statements/AbstractStatement.java]: This getType() method will confuse developers with getType() usages in the other parts of the code.

File: modules/ballerina-core/src/main/java/org/ballerinalang/model/statements/StatementType.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+* under the License.
+*/
+package org.ballerinalang.model.statements;
+
+/**
+ * Represents Ballerina Statement Type.
+ *
+ * @since 0.89
+ */
+public enum StatementType {
```
Comment (sameera-jayasoma) [inline modules/ballerina-core/src/main/java/org/ballerinalang/model/statements/StatementType.java]: I think we need to find a better classname here. This class contain both blocks and statements

File: modules/ballerina-core/src/main/java/org/ballerinalang/model/statements/AbstractStatement.java (hunk 1 above)
Comment (sameera-jayasoma) [inline modules/ballerina-core/src/main/java/org/ballerinalang/model/statements/AbstractStatement.java]: I think we should rename "Type" to "Kind"

File: modules/ballerina-core/src/main/java/org/ballerinalang/model/statements/StatementType.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    THROW(13),
+    FORK_JOIN(14),
+    TRANSACTION(15),
+    ABORT(16),
+    WORKER_INVOCATION(17),
+    WORKER_REPLY(18),
+    TRANSFORM(19),
+
+    // Blocks
+    ROOT_BLOCK(30),
```
Comment (sameera-jayasoma) [inline modules/ballerina-core/src/main/java/org/ballerinalang/model/statements/StatementType.java]: ROOT_BLOCK is also a misleading name :)

File: modules/ballerina-core/src/main/java/org/ballerinalang/model/statements/AbstractStatement.java (hunk 1 above)
Comment (hasithaa) [inline modules/ballerina-core/src/main/java/org/ballerinalang/model/statements/AbstractStatement.java]: +1

File: modules/ballerina-core/src/main/java/org/ballerinalang/model/statements/StatementType.java (hunk 3 above)
Comment (hasithaa) [inline modules/ballerina-core/src/main/java/org/ballerinalang/model/statements/StatementType.java]: Callable_Unit_Block should be ok i guess.

**Pull request merged**